cli/cmd/identity.go

```diff
@@ -0,0 +1,227 @@
+package cmd
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/grantae/certinfo"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var emitLog bool
+
+type certificate struct {
+	pod         string
+	container   string
+	Certificate []*x509.Certificate
+	err         error
+}
+
+type identityOptions struct {
+	pod       string
+	namespace string
+	selector  string
+}
+
+func newIdentityOptions() *identityOptions {
+	return &identityOptions{
+		pod:       "",
+		namespace: "",
+		selector:  "",
+	}
+}
+
+func newCmdIdentity() *cobra.Command {
+	emitLog = false
+	options := newIdentityOptions()
+
+	cmd := &cobra.Command{
+		Use:   "identity [flags] (PODS)",
+		Short: "Display the certificate(s) of one or more selected pod(s)",
+		Long: `Display the certificate(s) of one or more selected pod(s).
+		
+This command initiates a port-forward to a given pod or a set of pods and fetches the TLS certificate.
+		`,
+		Example: ` 
+ # Get certificate from pod foo-bar in the default namespace.
+ linkerd identity foo-bar
+		
+ # Get certificate from all pods with the label name=nginx
+ linkerd identity -l name=nginx
+		`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 0 && options.selector == "" {
+				return fmt.Errorf("Provide the pod name argument or use the selector flag")
+			}
+
+			pods, err := getPods(cmd.Context(), k8sAPI, options.namespace, options.selector, args)
+			if err != nil {
+				return err
+			}
+
+			resultCerts := getCertificate(k8sAPI, pods, k8s.ProxyAdminPortName, emitLog)
+			if len(resultCerts) == 0 {
+				fmt.Print("Could not fetch Certificate. Ensure that the pod(s) are meshed by running `linkerd inject`\n")
+				return nil
+			}
+			for i, resultCert := range resultCerts {
+				fmt.Printf("\nPOD %s (%d of %d)\n\n", resultCert.pod, i+1, len(resultCerts))
+				if resultCert.err != nil {
+					fmt.Printf("\n%s", resultCert.err)
```

Comment:
We should have a newline here after printing the error message or else (probably depending on your terminal) it indicate that it's missing. You can see that symbol at the end of output below:

```
❯ ./target/release/linkerd2-cli-git-a3b2afa6-linux-amd64 identity -n kube-system local-path-provisioner-6d59f47c7-7fnt7

POD local-path-provisioner-6d59f47c7-7fnt7 (1 of 1)


failed to find linkerd-admin port in linkerd-proxy container for given pod spec⏎ 
```

```suggestion
					fmt.Printf("\n%s\n", resultCert.err)
```